src/rtk/features/delegationRequest/delegationRequestSlice.ts

```diff
@@ -1,86 +1,111 @@
+import { getCookie } from '@/resources/Cookie/CookieMethods';
 
 export interface ApiDelegation {
   orgName: string;
   apiName: string;
 }
 
-export interface InitialState {
+export interface SliceState {
   loading: boolean;
   error: string;
   succesfulApiDelegations: ApiDelegation[];
   failedApiDelegations: ApiDelegation[];
+  batchPostSize: number;
+  batchPostCounter: number;
 }
 
 export interface DelegationRequest {
   apiIdentifier: string;
+  apiName: string;
   orgNr: string;
+  orgName: string;
 }
 
-const initialState: InitialState = {
+const initialState: SliceState = {
   loading: true,
   error: '',
-  succesfulApiDelegations: [
-    {
-      orgName: 'Org blabla',
-      apiName: 'Api blabla',
-    },
-  ],
-  failedApiDelegations: [
-    {
-      orgName: 'Org blabla',
-      apiName: 'Api blabla',
-    },
-  ],
+  succesfulApiDelegations: [],
+  failedApiDelegations: [],
+  batchPostSize: 0,
+  batchPostCounter: 0,
 };
 
-export const delegateApi = async (
-  state: InitialState,
-  apiList: DelegableApi[],
-  orgList: DelegableOrg[],
-) => {
-  for (let i = 0; i < apiList.length; i++) {
-    for (let j = 0; j < orgList.length; j++) {
-      const delegationRequest = {
-        apiIdentifier: apiList[i].id,
-        orgNr: orgList[j].orgNr,
-      };
-
-      const apiDelegation = {
-        apiName: apiList[i].apiName,
-        orgName: orgList[j].orgName,
-      };
-      try {
-        await postApiDelegation(delegationRequest);
-        state.succesfulApiDelegations.push(apiDelegation);
-      } catch {
-        state.failedApiDelegations.push(apiDelegation);
-      }
-    }
-  }
-};
-
-const postApiDelegation = createAsyncThunk(
-  'delegationRequestSlice/delegateApi',
-  async ({ apiIdentifier, orgNr }: DelegationRequest) => {
+export const postApiDelegation = createAsyncThunk(
+  'delegationRequestSlice/postApiDelegation',
+  async (delegationInfo: DelegationRequest) => {
+    const { apiIdentifier, apiName, orgNr, orgName }: DelegationRequest = delegationInfo;
+    const delegation: ApiDelegation = {
+      apiName,
+      orgName,
+    };
+    const altinnPartyId =
+      getCookie('AltinnPartyId') === null ? getCookie('AltinnPartyId') : '50067798';
     return await axios
-      .get('/accessmanagement/api/v1/1337/resources/maskinportenschema')
-      .then((response) => response.data)
+      .post(`/accessmanagement/api/v1/${altinnPartyId}/delegations/maskinportenschema`, {
+        to: [
+          {
+            id: 'urn:altinn:organizationnumber',
+            value: String(orgNr),
+          },
+        ],
+        rights: [
+          {
+            resource: [
+              {
+                id: 'urn:altinn:resourceregistry',
+                value: String(apiIdentifier),
+              },
+            ],
+          },
+        ],
+      })
+      .then((response) => {
+        return delegation;
+      })
       .catch((error) => {
-        console.log(error);
+        throw error;
```

Comment:
```suggestion
      throw new Error(String(error.response.status));
```

Reply:
Won't we lose all the other error information (other than just the status) like this?

I mean, we aren't really using the other info anyway, so it won't be a huge loss I guess, but why take the time to create a new Error when we've already got one we can just pass along? 😅